### PR TITLE
CP-42559: Add RBAC info to C#, Java, and C SDK method comments

### PR DESCRIPTION
**Pull request by danilo-delbusso**
The PowerShell SDK needs more work as we'd need to generate a valid `dll-Help.xml` file to properly comment cmdlets: https://learn.microsoft.com/en-us/powershell/scripting/developer/help/how-to-create-the-cmdlet-help-file?view=powershell-7.3

Comment based support for `Get-Help` is only available for modules written directly in PowerShell.

Bit out of scope for this, worth opening a separate ticket.

Example of the new docs:

C#

```csharp
/// <summary>
/// Create a new VTPM instance, and return its handle.
 /// Experimental. First published in 22.26.0.
/// </summary>
/// <param name="session">The session</param>
/// <param name="_vm">The VM reference the VTPM will be attached to</param>
/// <param name="_is_unique">Whether the VTPM must be unique</param>
/// <remarks>
/// Minimum allowed role: vm-admin
/// </remarks>
public static XenRef<VTPM> create(Session session, string _vm, bool _is_unique)
{
    return session.JsonRpcClient.vtpm_create(session.opaque_ref, _vm, _is_unique);
}
```

C
```c
/**
 * Create a new VTPM instance, and return its handle.
 * Minimum allowed role: vm-admin.
 */
extern bool
xen_vtpm_create(xen_session *session, xen_vtpm *result, xen_vm vm, bool is_unique);
```

Java

```java
/**
   * Create a new VTPM instance, and return its handle.
   * Minimum allowed role: vm-admin
   * Experimental. First published in 22.26.0.
   *
   * @param c The connection the call is made on
   * @param vM The VM reference the VTPM will be attached to
   * @param isUnique Whether the VTPM must be unique
   * @return The reference of the newly created VTPM
   * @throws BadServerResponse Thrown if the response from the server contains an invalid status.
   * @throws XenAPIException Thrown if the call failed.
   * @throws XmlRpcException Thrown if the result of an asynchronous call could not be parsed.
   */
public static VTPM create(Connection c, VM vM, Boolean isUnique) throws BadServerResponse, XenAPIException, XmlRpcException 
{
    String method_call = "VTPM.create";
    String session = c.getSessionReference();
    Object[] method_params = {Marshalling.toXMLRPC(session), Marshalling.toXMLRPC(vM), Marshalling.toXMLRPC(isUnique)};
    Map response = c.dispatch(method_call, method_params);
    Object result = response.get("Value");
    return Types.toVTPM(result);
}
```

### CODE REVIEW

File: ocaml/sdk-gen/java/main.ml
Comment (danilo-delbusso) [inline ocaml/sdk-gen/java/main.ml:291]: Could not find a better option for [javadoc](https://docs.oracle.com/javase/1.5.0/docs/tooldocs/windows/javadoc.html) style docs. So it's just in the generic method summary

File: ocaml/sdk-gen/common/CommonFunctions.ml
Comment (kc284) [inline ocaml/sdk-gen/common/CommonFunctions.ml:200]: This is probably a question for the Toosltack folk: Should we exclude internal roles?

I suppose this also goes for the API docs, which currently do not hide internal roles. For example, the mimimun role for `repository.set_name_label` is documented as `client-cert` - not sure how much sense this makes to the API users. (Note that on the UI we do hide internal roles).

Comment (danilo-delbusso) [reply ocaml/sdk-gen/common/CommonFunctions.ml:200]: cc @psafont @lindig

Comment (psafont) [reply ocaml/sdk-gen/common/CommonFunctions.ml:200]: I think it makes sense as a policy to hide internal roles as a policy. Might as well start now here

**Review by kc284 — CHANGES_REQUESTED** [review]
Ok, based on Pau's reply, let's exclude internal roles.

File: ocaml/idl/datamodel_roles.ml
Comment (psafont) [inline ocaml/idl/datamodel_roles.ml:81]:
```suggestion
  match List.assoc_opt role_name_label role_is_internal_map with
  | Some is_internal ->
      is_internal
  | None ->
      failwith
        (Printf.sprintf
           "Check Datamodel_roles.role_is_internal: there's no is_internal field \
            set for role %s"
           role_name_label
        )
```
It's usually more efficient to not raise exceptions to immediately capture it and use options instead

**Review by psafont — DISMISSED** [review]

File: ocaml/sdk-gen/common/CommonFunctions.ml
Comment (kc284) [inline ocaml/sdk-gen/common/CommonFunctions.ml:206]: I was diffing the generated code:
- I see `Not Applicable` being attached to all `get_all_records` calls, for which I would expect the role to be `read-only`.
- Toolstack can correct me if I'm wrong, but for the calls that previously had an internal role, shouldn't the docs show the next public available one, i.e. `read-only`?

**Review by kc284 — CHANGES_REQUESTED** [review]
As above.

**Review by kc284 — APPROVED** [review]

**Review by kc284 — APPROVED** [review]

**Comment by kc284** [comment]
@psafont Danilo dismissed your review due to adding one more commit after you had approved, could you please have one more look - thanks.

**Pull request merged**
